health-services/project-factory/src/server/utils/microplanUtils.ts

```
@@ -0,0 +1,431 @@
+        if (boundary?.length > maxBoundaryDataLength) {
+          maxBoundaryDataLength = boundary?.length;
+        }
+      }
+
+      // If boundary data has sufficient depth, set `villageIndex` to the second-to-last level
+      if (maxBoundaryDataLength >= 2) {
+        villageIndex = maxBoundaryDataLength - 2;
+      }
+    }
+    // Filter out boundaries that don't have the `villageIndex`
+    boundaryData = boundaryData.filter((boundary: any) => boundary?.[villageIndex]);
```

Biome

[error] Reassigning a function parameter is confusing.

The parameter is declared here:

Use a local variable instead.

(lint/style/noParameterAssign)